webpack/assets/javascripts/react_app/IntergrationHelper.js

```diff
@@ -0,0 +1,19 @@
+import { applyMiddleware, combineReducers, createStore } from 'redux';
+import thunk from 'redux-thunk';
+
+export function flushAllPromises() {
+  return new Promise(resolve => setTimeout(() => resolve(), 100));
```

Comment:
Why don't you use the `setImmediate` like the post propose to?

`setImmediate` should get called when the javascript callstack is getting empty.
It is till risky, but somehow make sense when waiting for all the async actions to get resolved.